server/services/timelineService.test.ts

```diff
@@ -52,48 +53,141 @@ describe('timelineService', () => {
       const timelineResponse: TimelineResponse = {
         reference: '6add2455-30f1-4b3e-a23e-1baf2d761e8f',
         prisonNumber: 'A1234BC',
         events: [
           {
             reference: 'f49a3412-df7f-41d2-ac04-ffd35e453af4',
-            sourceReference: '32',
-            eventType: 'ACTION_PLAN_CREATED',
+            sourceReference: '1211013',
+            eventType: 'PRISON_ADMISSION',
+            prisonId: 'MDI',
+            actionedBy: 'system',
+            timestamp: '2023-08-01T10:46:38.565Z',
+            correlationId: '6457a634-6dbe-4179-983b-74e92883232c',
+            contextualInfo: undefined,
+            actionedByDisplayName: undefined,
+          },
+          {
+            reference: '97cf372e-009f-4781-83d9-e4cc74ee321e',
+            sourceReference: '3a60b143-dfe3-49a7-9cba-31c2b8a7fa8b',
+            eventType: 'GOAL_CREATED',
             prisonId: 'MDI',
             actionedBy: 'RALPH_GEN',
             timestamp: '2023-09-01T10:46:38.565Z',
             correlationId: '847aa5ad-2068-40e1-aec0-66b19007c494',
-            contextualInfo: '',
+            contextualInfo: 'Learn French',
             actionedByDisplayName: 'Ralph Gen',
           },
+        ],
+      }
+      educationAndWorkPlanClient.getTimeline.mockResolvedValue(timelineResponse)
+      const expectedFilteredTimelineResponse: TimelineResponse = {
+        reference: '6add2455-30f1-4b3e-a23e-1baf2d761e8f',
+        prisonNumber: 'A1234BC',
+        events: [
           {
-            reference: 'cd98ea4c-b415-48d9-a600-9068cefe65e4x',
-            sourceReference: '33bc1045-7368-47c4-a261-4d616b7b51b9',
+            reference: 'f49a3412-df7f-41d2-ac04-ffd35e453af4',
+            sourceReference: '1211013',
+            eventType: 'PRISON_ADMISSION',
+            prisonId: 'MDI',
+            actionedBy: 'system',
+            timestamp: '2023-08-01T10:46:38.565Z',
+            correlationId: '6457a634-6dbe-4179-983b-74e92883232c',
+            contextualInfo: undefined,
+            actionedByDisplayName: undefined,
+          },
+          {
+            reference: '97cf372e-009f-4781-83d9-e4cc74ee321e',
+            sourceReference: '3a60b143-dfe3-49a7-9cba-31c2b8a7fa8b',
             eventType: 'GOAL_CREATED',
             prisonId: 'MDI',
             actionedBy: 'RALPH_GEN',
-            timestamp: '2023-09-01T10:47:38.565Z',
-            correlationId: '246aa049-c5df-459d-8231-bdeab3936d0f',
-            contextualInfo: '',
+            timestamp: '2023-09-01T10:46:38.565Z',
+            correlationId: '847aa5ad-2068-40e1-aec0-66b19007c494',
+            contextualInfo: 'Learn French',
             actionedByDisplayName: 'Ralph Gen',
           },
+        ],
+      }
+      const timeline: Timeline = aValidTimeline()
+      mockedTimelineMapper.mockReturnValue(timeline)
+
+      // When
+      await timelineService.getTimeline(prisonNumber, userToken, username)
```

Comment:
Do we no longer want to capture the return value from the method as `actual` and then make some kind of assertion on it? 🤔

Reply:
We could do, but it will be whatever's return from the mocked mapper: 
```
mockedTimelineMapper.mockReturnValue(timeline)
```
I guess I could add it for regression purposes, but the point of this test is to make sure that the mapper is called with the correct filtered items.

Reply:
I've added an assertion for the returned value.